src/main/scala/junctions/asyncqueue.scala

```diff
@@ -66,9 +76,16 @@ class AsyncQueueSink[T <: Data](gen: T, depth: Int, sync: Int, clockIn: Clock, r
   // On an FPGA, only one input changes at a time => mem updates don't cause glitches
   // The register only latches when the selected valued is not being written
   val index = if (depth == 1) UInt(0) else ridx(bits-1, 0) ^ (ridx(bits, bits) << (bits-1))
-  io.deq.bits  := RegEnable(io.mem(index), valid && !reset)
-  io.deq.valid := RegNext(valid, Bool(false))
-  io.ridx := RegNext(ridx, UInt(0))
+  // This register does not NEED to be reset, as its contents will not
+  // be considered unless deq_reg is true.
+  io.deq.bits  := RegEnable(io.mem(index), valid) 
```

Comment:
The !reset was there as a work-around for the fact that I did not have async reset. I didn't want it latching garbage. You were right to remove it.